### PR TITLE
gh-99108: Import MD5 and SHA1 from HACL*

### PR DESCRIPTION
**Pull request by msprotz**
Continuing the import of HACL* algorithms for hashlib, this time for legacy MD5 and SHA1.

* Issue: gh-99108

### CODE REVIEW

**Comment by msprotz** [comment]
(I still need to fixup `python_hacl_namespaces.h` but first want to make sure this PR works.)

**Comment by msprotz** [comment]
@gpshead these are just build errors, apparently I didn't export the right CFLAGS somewhere

before I dig further into this, would you like all of the hacl files to be bundled into a single libhacl.a? or is it ok to rely on the existing build system to build single .o files for the non-sha2 algorithms?

**Comment by gpshead** [comment]
> before I dig further into this, would you like all of the hacl files to be bundled into a single libhacl.a? or is it ok to rely on the existing build system to build single .o files for the non-sha2 algorithms?

I _think_ what you've done in ths PR should work fine. one .c file for each of md5 and sha1, no specific need for a library.

from a build simplicity perspective with the way we have things set up in our plethora of build configs, the lesson from the sha2 stuff was not to reuse the same .c/.o/.a with non-static symbols in multiple extension modules. _(i may not have needed the .a for sha2 in the end but i decided to leave it in place when merging sha256 and sha512 into sha2 - if we don't need it setup that way in the end it could be undone later to just be the .c file as these are)_

**Comment by msprotz** [comment]
Ok looks like we're good now. I'll get started on SHA3 while I wait for this one to land.

**Comment by msprotz** [comment]
I'm not sure what to make of the buildbot run. Three failures are in test_zipfile, another one in test_asyncio, all seem unrelated to this PR.

**Review by gpshead — APPROVED** [review]

File: Modules/Setup [hunk 1]
```diff
@@ -163,8 +163,8 @@ PYTHONPATH=$(COREPYTHONPATH)
 
 # hashing builtins
 #_blake2 _blake2/blake2module.c _blake2/blake2b_impl.c _blake2/blake2s_impl.c
-#_md5 md5module.c
-#_sha1 sha1module.c
+#_md5 md5module.c -I$(srcdir)/Modules/_hacl/include _hacl/libHacl_Hash_MD5.c
+#_sha1 sha1module.c -I$(srcdir)/Modules/_hacl/include _hacl/libHacl_Hash_SHA1.c
```
Comment (asottile) [inline Modules/Setup:167]: these have the wrong filename :(

Comment (msprotz) [reply Modules/Setup:167]: https://github.com/python/cpython/pull/102183

**Pull request merged**